src/CustomRouting.php

```diff
@@ -0,0 +1,12 @@
+<?php declare(strict_types=1);
+
+namespace ElasticScoutDriver;
+
+/**
+ * Implement this interface to enable custom elasticsearch routing functionality.
+ * https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-routing-field.html
+ */
+interface CustomRouting
```

Comment:
I'd prefer to use a trait rather than an interface to be consistent with the current approach:

```php
trait ShardRouting
{
    public function getRoutingPath(): string
    {
         return $this->getScoutKeyName();
    }
}
```

Reply:
One more thing, `ShardRouting` trait must be in Elastic Scout Driver Plus package, because this is not a compatible with Scout feature.